app/presenters/mail_presenter.rb

```diff
@@ -8,30 +8,47 @@ def initialize(mail, account = nil)
   end
 
   def subject
-    encode_to_unicode(@mail.subject || '')
+    encode_to_unicode(@mail.subject)
   end
 
   def text_content
-    @decoded_text_content ||= encode_to_unicode(text_part&.decoded || decoded_message || '')
+    @decoded_text_content = select_body(text_part) || ''
+    encoding = @decoded_text_content.encoding
+
+    body = EmailReplyTrimmer.trim(@decoded_text_content)
 
     return {} if @decoded_text_content.blank?
 
     @text_content ||= {
-      full: @decoded_text_content,
-      reply: extract_reply(@decoded_text_content)[:reply],
-      quoted: extract_reply(@decoded_text_content)[:quoted_text]
+      full: select_body(text_part),
+      reply: @decoded_text_content,
+      quoted: body.force_encoding(encoding).encode('UTF-8')
     }
   end
 
+  def select_body(mail_part)
+    return '' unless mail_part
+
+    decoded = encode_to_unicode(mail_part.decoded)
+
+    if mail.text_part
+      decoded
+    elsif (mail.content_type || '').include? 'text/html'
+      ::HtmlParser.parse_reply(decoded)
```

Comment:
if do `parse_reply` on HTML emails will it strip off all the HTML and convert it into a text email ?

Reply:
No, it parses the HTML tags and applies them. It doesn't convert it to the text.